skfp/fingerprints/atom_pair.py

```diff
@@ -206,14 +216,24 @@ def _calculate_fingerprint(
             countSimulation=self.count_simulation,
         )
         if self.count:
-            X = [
+            Y = [
                 gen.GetCountFingerprintAsNumPy(mol, confId=conf_id)
                 for mol, conf_id in zip(X, conf_ids)
             ]
         else:
-            X = [
+            Y = [
                 gen.GetFingerprintAsNumPy(mol, confId=conf_id)
                 for mol, conf_id in zip(X, conf_ids)
             ]
 
-        return csr_array(X) if self.sparse else np.array(X)
+        if self.normalize:
+            if self.count:
+                Y = [self._scale_by_hac(fp, mol) for fp, mol in zip(Y, X)]
+            else:
+                warnings.warn("Scaling by HAC can only be applied to count vectors.")
```

Comment:
Maybe add "No HAC scaling will be applied" as a second sentence, to be more explicit?